### PR TITLE
Fix for JENKINS-53739 - user.name

### PR DESCRIPTION
**Pull request by pjanouse**

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/proccleaner/PsCleaner.java [hunk 1]
```diff
@@ -64,7 +69,34 @@ public void clean(CleanRequest request) throws IOException, InterruptedException
         }
 
         try {
-            killer.kill(System.getProperty("user.name", ""), request.getListener().getLogger());
+            String osName = System.getProperty( "os.name" ).toLowerCase();
+            String className = null;
+            String methodName = "getUsername";
+            String user = null;
+
+            // https://issues.jenkins-ci.org/browse/JENKINS-53739
```
Comment (vjuranek) [inline src/main/java/org/jenkinsci/plugins/proccleaner/PsCleaner.java:77]: all the stuff bellow won't work on e.g. IBM java, but I guess good for now and we can fix it when some issue for IBM java is looged :-)

**Comment by vjuranek** [comment]
Merged, thanks! Released as `2.2`

File: src/main/java/org/jenkinsci/plugins/proccleaner/PsCleaner.java (hunk 1 above)
Comment (pjanouse) [reply src/main/java/org/jenkinsci/plugins/proccleaner/PsCleaner.java:77]: Sure, but in that case the old way is used or I've overlooked sth.?

Comment (vjuranek) [reply src/main/java/org/jenkinsci/plugins/proccleaner/PsCleaner.java:77]: no, you are right, I just pointed out that it doesn't fix the issue for all JDKs.

Comment (pjanouse) [reply src/main/java/org/jenkinsci/plugins/proccleaner/PsCleaner.java:77]: JNA (Core uses it too) provides a way how to do similar thing - probably more portable... WDYT?

Comment (vjuranek) [reply src/main/java/org/jenkinsci/plugins/proccleaner/PsCleaner.java:77]: depends on you if you want to spend time with it. IMHO IBM JDK is not used very often and I consider it as a very minor bug and low priority

**Pull request merged**